app/components/autocomplete/channel_mention_item/index.js

```diff
@@ -6,15 +6,23 @@ import {connect} from 'react-redux';
 import {getChannel} from 'mattermost-redux/selectors/entities/channels';
 
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+import {General} from 'mattermost-redux/constants';
 
 import ChannelMentionItem from './channel_mention_item';
 
 function mapStateToProps(state, ownProps) {
     const channel = getChannel(state, ownProps.channelId);
+    let displayName = channel.display_name;
+
+    // Bypassing the channel display name generation in DMs and GMs
+    if (channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL) {
+        displayName = state.entities.channels.channels[ownProps.channelId].display_name;
```

Comment:
Could you use a selector to get channel here? so if we ever change state object we don't run into issues?

Reply:
2/5 to use a selector